src/pages/chapter/[chapterAbbreviation]/lineage/index.tsx

```diff
@@ -0,0 +1,144 @@
+}: {
+  chapterAbbreviation: string;
+  chapterName: string;
+  lineage: LineProps[];
+}) {
+  const [_lineage, setLineage] = useState(lineage);
+  const [hasMore, setHasMore] = useState(true);
+
+  if (!lineage.length) {
+    return <div>Lineage Not Found ...</div>;
+  }
+
+  const addNewLines = async (): Promise<void> => {
+    const [_chapterName, additionalLines, totalLines] = await getChapterLineage(
+      chapterAbbreviation,
+      _lineage.length.toString(),
+      "10"
+    );
+    setLineage((_lineage) => [..._lineage, ...additionalLines]);
+    setHasMore(totalLines > _lineage.length ? true : false);
+  };
+
+  return (
+    <div>
+      <Head>
+        <title>{`${chapterAbbreviation} Chapter Lineage`}</title>
+      </Head>
+      <PageTemplate>
+        <div className="md:container md:mx-auto mt-12 min-h-screen">
+          <span className="font-bold text-xl">{chapterName}</span>
+          <div className="font-bold text-heading-3">Lineage</div>
+          <InfiniteScroll
+            dataLength={_lineage ? _lineage.length : 0}
+            next={() => addNewLines()}
+            hasMore={hasMore}
+            loader={<h4>Loading...</h4>}
+            endMessage={
+              <p className="text-center text-[8px] mt-8">
+                <b>{`You've reached the end of ${chapterAbbreviation}'s Chapter Lineage`}</b>
+              </p>
+            }
+          >
+            <div className="mt-4 grid md:items-center md:grid-cols-4 md:gap-4 gap-y-4">
+              {_lineage.map((line: any, index: number) => (
+                <Link
+                  href={{
+                    pathname: `/chapter/${chapterAbbreviation}/line/${line.id}`,
+                  }}
+                  key={v4()}
+                >
+                  <Line
+                    key={line.id}
+                    term={line.term}
+                    year={line.year}
+                    shipName={line.ship_name}
+                  />
+                </Link>
+              ))}
+            </div>
+          </InfiniteScroll>
+        </div>
+      </PageTemplate>
+    </div>
+  );
+}
+
+export const getServerSideProps: GetServerSideProps<{
+  chapterAbbreviation: string;
+  chapterName: string;
+  lineage: LineProps[];
+}> = async ({ query }) => {
+  let { chapterAbbreviation } = query;
+  chapterAbbreviation = (chapterAbbreviation as string).toUpperCase();
+
+  const [chapterName, lineage, totalLines] = await getChapterLineage(
+    chapterAbbreviation,
+    "0",
+    "10"
+  );
+
+  if (!chapterName) {
+    return {
+      notFound: true,
+    };
+  }
+
+  return {
+    props: { chapterAbbreviation, chapterName, lineage },
+  };
+};
+
+async function getChapterLineage(
+  chapterAbbreviation: string,
+  start: string,
+  limit: string
+): Promise<[string | undefined, LineProps[], number]> {
+  const [jsonChapterLineage, error] = await getChapterLines(
+    chapterAbbreviation,
+    start,
+    limit
+  );
+
+  if (error) {
+    console.log(error);
```

Comment:
We gotta find another way to display errors.

Reply:
I with ya